packages/flutter/lib/src/material/card_theme.dart

```diff
@@ -30,13 +30,191 @@ import 'theme.dart';
 ///
 ///  * [ThemeData], which describes the overall theme information for the
 ///    application.
-@immutable
-class CardTheme with Diagnosticable {
+class CardTheme extends InheritedWidget with Diagnosticable {
 
   /// Creates a theme that can be used for [ThemeData.cardTheme].
   ///
   /// The [elevation] must be null or non-negative.
   const CardTheme({
+    super.key,
+    Clip? clipBehavior,
+    Color? color,
+    Color? surfaceTintColor,
+    Color? shadowColor,
+    double? elevation,
+    EdgeInsetsGeometry? margin,
+    ShapeBorder? shape,
+    CardThemeData? data,
+    Widget? child,
+  }) : assert(
+        data == null ||
+        (clipBehavior ??
+        color ??
+        surfaceTintColor ??
+        shadowColor ??
+        elevation ??
+        margin ??
+        shape) == null),
+      assert(elevation == null || elevation >= 0.0),
+      _data = data,
+      _clipBehavior = clipBehavior,
+      _color = color,
+      _surfaceTintColor = surfaceTintColor,
+      _shadowColor = shadowColor,
+      _elevation = elevation,
+      _margin = margin,
+      _shape = shape,
+      super(child: child ?? const SizedBox());
+
+  final CardThemeData? _data;
+  final Clip? _clipBehavior;
+  final Color? _color;
+  final Color? _surfaceTintColor;
+  final Color? _shadowColor;
+  final double? _elevation;
+  final EdgeInsetsGeometry? _margin;
+  final ShapeBorder? _shape;
+
+  /// Overrides the default value for [Card.clipBehavior].
+  ///
+  /// This property is obsolete: please use the [CardThemeData.clipBehavior]
```

Comment:
> a fix 

Dart fix can't help with this one unfortunately. We looked at it a bunch of ways, but since the class CardTheme itself will still exist undeprecated, we can't auto migrate it to CardThemeData.